brig/cmd/brig/commands/build_list_test.go

```diff
@@ -0,0 +1,268 @@
+package commands
+
+import (
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	stubProject1ID = "brigade-544b459e6ad7267e7791c4f77bfd1722a15e305a22cf9d3c60c5be"
+	stubProject2ID = "brigade-544b459e6ad7267e7791c4f77bfd1722a15e305a22cf9d3c60c5bf"
+	stubBuild1ID   = "01cxmy71nbq7nasvth8pva1s21"
+	stubBuild2ID   = "01cxmy71nbq7nasvth8pva1s22"
+	stubBuild3ID   = "01cxmy71nbq7nasvth8pva1s23"
+)
+
+var (
+	stubDT1Start     = time.Now().Add(-5 * time.Minute)
+	stubTimeDT1Start = metav1.NewTime(stubDT1Start)
+	stubDT1End       = time.Now().Add(-2 * time.Minute)
+
+	stubDT2Start     = time.Now().Add(-time.Minute)
+	stubTimeDT2Start = metav1.NewTime(stubDT2Start)
+
+	stubDT3Start     = time.Now().Add(-3 * time.Minute)
+	stubTimeDT3Start = metav1.NewTime(stubDT3Start)
+	stubDT3End       = time.Now().Add(-time.Minute)
+)
+
+func TestGetEmptyBuildList(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	bls, err := getBuilds("", client, 0)
+	if err != nil {
+		t.Error(err)
+	}
+	if len(bls) != 0 {
+		t.Error("Error in getBuilds for no project(s)")
+	}
+
+	bls, err = getBuilds("", client, 5)
+	if err != nil {
+		t.Error(err)
+	}
+	if len(bls) != 0 {
+		t.Error("Error in getBuilds for no project(s) with --count 5")
+	}
+}
+
+// TestGetBuildList tests the command `brig build list`
+func TestGetBuildList(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	createFakeBuilds(t, client)
+	bls, err := getBuilds("", client, 0)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(bls) != 3 {
+		t.Error("Error in getBuilds for all projects")
+	}
+
+	if bls[0].since != "???" || bls[0].ID != stubBuild2ID {
+		t.Error("Error in build2 time")
+	}
+	if bls[1].since != "1m" || bls[1].ID != stubBuild3ID {
+		t.Error("Error in build3 time")
+	}
+	if bls[2].since != "2m" || bls[2].ID != stubBuild1ID {
+		t.Error("Error in build1 time")
+	}
+}
+
+// TestGetBuildListWithProject tests the command `brig build list projectID`
+func TestGetBuildListWithProject(t *testing.T) {
+	client := fake.NewSimpleClientset()
+	createFakeBuilds(t, client)
+	bls, err := getBuilds(stubProject1ID, client, 0)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if len(bls) != 2 {
+		t.Errorf("Error in getBuilds for project %s", stubProject1ID)
+	}
+
+	if bls[0].since != "???" || bls[0].ID != stubBuild2ID {
+		t.Error("Error in build2")
+	}
+	if bls[1].since != "2m" || bls[1].ID != stubBuild1ID {
+		t.Error("Error in build1")
+	}
+
+}
+
+// TestGetBuildListWithProject tests the command `brig build list --count 2`
```

Comment:
Mismatch here with comment and actual function name.

Reply:
thanks!